libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -110,5 +110,37 @@ public bool Read(int i, ref byte* ptr, byte* end)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int GetInt(int i)
             => ParseUtils.ReadInt(ref Unsafe.AsRef<ArgSlice>(bufferPtr + i));
+
+        /// <summary>
+        /// Try to get int argument at the given index
+        /// </summary>
+        /// <returns>True if integer parsed successfully</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool TryGetInt(int i, out int value)
```

Comment:
Could be helpful to identify bugs to have an assert that the index is within count